sds_data_manager/lambda_code/SDSCode/pipeline_lambdas/batch_starter.py

```diff
@@ -366,16 +362,26 @@ def try_to_submit_job(
     logger.info(
         f"Wrote job INPROGRESS to Processing Jobs Table with id: {processing_job.id}"
     )
-
-    # Reformat the upstream dependencies from dependency call to match
-    # what batch job expects.
     # TODO: do we need a reprocessing column to indicate if this is a reprocessing job?
 
-    # If upstream dependencies are a ProcessingInputCollection, serialize them into a
-    # string. Otherwise, it should be a string representing the filename of a JSON file
-    # containing the serialized upstream dependencies.
-    if isinstance(upstream_dependencies, ProcessingInputCollection):
-        upstream_dependencies = upstream_dependencies.serialize()
+    # Serialize the upstream dependencies and write them to a JSON file. The Imap
+    # processing code will read the JSON file and deserialize the dependencies. This is
+    # to avoid passing a large string through the batch job command line.
+    # TODO replace CadenceFilePath with DependencyFilePath after new imap-data-access
+    # release
+    dependency_file = CadenceFilePath.generate_from_inputs(
+        instrument=instrument,
+        data_level=data_level,
+        descriptor=descriptor,
+        start_time=start_date.strftime("%Y%m%d"),
+        version=version,
+        extension="json",
+    )
+    dependency_file_path = Path(dependency_file.construct_path())
```

Comment:
Isn't this already a Path object?
```suggestion
    dependency_file_path = dependency_file.construct_path()
```